server/src/main/java/io/crate/netty/NettyBootstrap.java

```diff
@@ -73,4 +82,32 @@
             }
         });
     }
+
+    public static Class<? extends Channel> clientChannel() {
+        if (Epoll.isAvailable()) {
+            return EpollSocketChannel.class;
+        } else {
+            return NioSocketChannel.class;
+        }
+    }
+
+    public static Class<? extends ServerSocketChannel> serverChannel() {
+        if (Epoll.isAvailable()) {
+            return EpollServerSocketChannel.class;
+        } else {
+            return NioServerSocketChannel.class;
+        }
+    }
+
+    public static ServerBootstrap newServerBootstrap(Settings settings, EventLoopGroup eventLoopGroup) {
```

Comment:
Looks like this is only used for the PG protocol some maybe rename to a better name? E.g. `newPostgresServerBootstrap`.

Any reason not to move the bootstraps for `http`+ `transport` also to this class to have all bootstrap codes nearby?

Reply:
Renaming to `newPostgresServerBootstrap` would raise the question why the server bootstrap is Postgres specific and it kinda isn't. If we clean up http + transport we could probably re-use it there as well, I just didn't want to go that far. 

I only wanted the channel methods for the pg-client work I'm doing for CCR. Moving this here was a side effect of the rename to avoid having similar named classes (`CrateChannelBootstrapFactory` and `NettyBootstrap`)

Reply:
>  If we clean up http + transport we could probably re-use it there as well,

Yeah this was popping up at my mind, but seems like at least the settings differ. But shouldn't prevent merging it.
I've just was a bit confused at first why this is only used inside the PG transport.